### PR TITLE
optimize maxAllowedPacket config fix: #292 #281

### LINKED ISSUE #292: Refactor: Optimize `maxAllowedPacket` config.
**What would you like to be added**:

**Why is this needed**:

- rename `maxAllowedPacket` to `max_allowed_packet`.
- extract public configuration content.

### PR DESCRIPTION
**Pull request by Penglq**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #292 #281 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/config/model.go [hunk 1]
```diff
@@ -21,12 +21,15 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"golang.org/x/text/cases"
```
Comment (georgehao) [inline pkg/config/model.go:24]: format import

File: pkg/boot/discovery.go [hunk 2]
```diff
@@ -163,6 +165,14 @@ func (fp *discovery) GetConfigCenter() *config.Center {
 	return fp.c
 }
 
+func (fp *discovery) GetClusterObject(ctx context.Context, cluster string) (*config.DataSourceCluster, error) {
+	dataSourceCluster, ok := fp.loadCluster(cluster)
+	if !ok {
+		return nil, nil
```
Comment (georgehao) [inline pkg/boot/discovery.go:171]: maybe need return a err?

File: pkg/config/model.go [hunk 3]
```diff
@@ -141,17 +145,40 @@ type (
...
+		sBuff.WriteString(pm.Lcfirst(pm.Case2Camel(k)))
 		sBuff.WriteString("=")
 		sBuff.WriteString(v)
 		sBuff.WriteString("&")
 	}
 	return strings.TrimRight(sBuff.String(), "&")
 }
 
+// Case2Camel underline to camel
+func (pm *ParametersMap) Case2Camel(name string) string {
```
Comment (georgehao) [inline pkg/config/model.go:168]: rename the func name. how about CamelCase() ?

File: pkg/config/model.go [hunk 4]
```diff
@@ -141,17 +145,40 @@ type (
...
+	name = strings.Replace(name, "_", " ", -1)
+	name = cases.Title(language.English, cases.NoLower).String(name)
+	return strings.Replace(name, " ", "", -1)
+}
+
+// Lcfirst lowercase letter
+func (pm *ParametersMap) Lcfirst(str string) string {
+	for i, v := range str {
+		return string(unicode.ToLower(v)) + str[i+1:]
+	}
```
Comment (georgehao) [inline pkg/config/model.go:178]:
```go
return string(unicode.ToLower(str[0])) + str[1:])
```

File: pkg/mysql/client.go [hunk 5]
```diff
@@ -24,10 +24,12 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/arana-db/arana/pkg/constants"
```
Comment (georgehao) [inline pkg/mysql/client.go:27]: format code

File: pkg/mysql/client.go [hunk 6]
```diff
@@ -1164,7 +1244,7 @@ func (conn *BackendConnection) ReadColumnDefinitions() ([]proto.Field, error) {
 }
 
 // ExecuteWithWarningCountIterRow is for fetching results and a warning count
-// Note: In a future iteration this should be abolished and merged into the Execute API.
+// Note: In a future iteration this should be abolished and merged into to Execute API.
```
Comment (georgehao) [inline pkg/mysql/client.go:1247]: // Note: In a future iteration this should be abolished and merged into the Executed API.

maybe this ?

File: pkg/boot/discovery.go [hunk 7]
```diff
@@ -89,6 +89,8 @@ type Discovery interface {
 
 	// ListClusters lists the cluster names.
 	ListClusters(ctx context.Context) ([]string, error)
+	// GetClusterObject returns the dataSourceCluster object
+	GetClusterObject(ctx context.Context, cluster string) (*config.DataSourceCluster, error)
```
Comment (dongzl) [inline pkg/boot/discovery.go:93]: Suggest rename the method, `GetClusterObject`  is confused with `GetCluster`.

`GetDataSourceCluster `  directly  or  other.

File: pkg/boot/boot.go [hunk 8]
```diff
@@ -87,10 +87,17 @@ func Boot(ctx context.Context, provider Discovery) error {
 
 func buildNamespace(ctx context.Context, provider Discovery, cluster string) (*namespace.Namespace, error) {
 	var (
-		groups []string
-		err    error
+		clusterObjects *config.DataSourceCluster
```
Comment (dongzl) [inline pkg/boot/boot.go:90]: Suggest  named `cluster`.

File: pkg/mysql/client.go (hunk 5 above)
Comment (dongzl) [inline pkg/mysql/client.go:27]: format import.

**Review by dongzl — CHANGES_REQUESTED** [review]

File: pkg/runtime/optimize/optimizer.go [hunk 9]
```diff
@@ -23,6 +23,8 @@ import (
 )
 
 import (
+	"github.com/arana-db/parser/ast"
+
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/optimizer.go:27]: reformat imports

**Review by jjeffcaii — APPROVED** [review]
LGTM

File: pkg/config/model.go [hunk 10]
```diff
@@ -141,17 +146,37 @@ type (
...
 
+// Camel underline to camel
+func (pm *ParametersMap) Camel(name string) string {
+	name = strings.Replace(name, "_", " ", -1)
+	name = cases.Title(language.English, cases.NoLower).String(name)
+	return strings.Replace(name, " ", "", -1)
+}
+
+// LowercaseFirstLetter lowercase letter
+func (pm *ParametersMap) LowercaseFirstLetter(str string) string {
```
Comment (dongzl) [inline pkg/config/model.go:176]: Suggest `LowerCase`.

**Review by dongzl — CHANGES_REQUESTED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**